packages/relay/src/lib/eth.ts

```diff
@@ -2371,6 +2372,28 @@ export class EthImpl implements Eth {
     }
   }
 
+  /**
+   * This method retrieves the contract address from the receipt response.
+   * If the contract creation is via a system contract, it handles the system contract creation.
+   * If not, it returns the address from the receipt response.
+   *
+   * @param {any} receiptResponse - The receipt response object.
+   * @returns {string} The contract address.
+   */
+  private getContractAddressFromReceipt(receiptResponse: any): string {
+    const isCreationViaSystemContract = constants.HTS_CREATE_FUNCTIONS_SELECTORS.includes(
+      receiptResponse.function_parameters.substring(0, constants.FUNCTION_SELECTOR_CHAR_LENGTH),
+    );
+
+    if (!isCreationViaSystemContract) {
+      return receiptResponse.address;
+    }
+
+    // Handle system contract creation
+    const tokenAddress = receiptResponse.call_result.substring(90);
```

Comment:
*nit: This makes sense and is explained in the design document, but it may confuse developers who revisit this in the future as they might not be aware of the documentation. Perhaps we can comments explaining the context for substringing from the 90th position or include a link to the relevant section of the design document.

Reply:
Are you referring to this section https://github.com/hashgraph/hedera-json-rpc-relay/blob/main/docs/design/hts_address_tx_receipt.md#2-extract-the-token-address-from-the-call_result-field-in-the-transaction-response-the-extractions-can-happen-with-brute-force-since-we-know-what-the-function-response-is-beforehand-and-we-know-the-last-40-characters-of-the-response-are-the-token-address?
It's not clear to me since design doc notes `const tokenAddress = receiptResponse.call_result.substring(receiptResponse.call_result.length - 40);` not 90th character.
Last 40 chars seems more general than 90th char 🤷🏾‍♂️ 

Not blocking for this PR but maybe in follow up a clearer summary description here with a note to the design doc is better.

Reply:
Yeah you're right I also needed to do some math to understand why it was set at the 90th. I agree that we should revert it to `- 40` as it's less confusing and also 40 is more general as 40 digits equal 20 bytes.

Reply:
@Nana-EC @quiet-node will upgrade this in a following pr